### ISSUE TITLE
gdl-1.0.1 FTBFS looking for -ltirpc

### ISSUE THREAD

**Issue by nieder** [fink/fink-distributions#866]
On 10.14.6 with Xcode11.3.1 and Xcode.app 11.3.1
```
-- Found Threads: TRUE  
-- Looking for xdr_uint64_t in tirpc
-- Looking for xdr_uint64_t in tirpc - not found
CMake Error at CMakeLists.txt:371 (message):
  RPC support is mandatory.
```
CMakeError.log has this:
```
Determining if the function xdr_uint64_t exists in the tirpc failed with the following output:
Change Dir: /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/build/CMakeFiles/CMakeTmp

Run Build Command(s):/sw/bin/make -f Makefile cmTC_3ad6e/fast && /sw/bin/make  -f CMakeFiles/cmTC_3ad6e.dir/build.make CMakeFiles/cmTC_3ad6e.dir/build
make[1]: Entering directory '/sw/build.build/gdl-1.0.1-1/gdl-1.0.1/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_3ad6e.dir/CheckFunctionExists.c.o
/sw/var/lib/fink/path-prefix-clang/cc   -DCHECK_FUNCTION_EXISTS=xdr_uint64_t -fPIE -MD -MT CMakeFiles/cmTC_3ad6e.dir/CheckFunctionExists.c.o -MF CMakeFiles/cmTC_3ad6e.dir/CheckFunctionExists.c.o.d -o CMakeFiles/cmTC_3ad6e.dir/CheckFunctionExists.c.o -c /sw/share/cmake-3.20/Modules/CheckFunctionExists.c
Linking C executable cmTC_3ad6e
/sw/bin/cmake -E cmake_link_script CMakeFiles/cmTC_3ad6e.dir/link.txt --verbose=1
/sw/var/lib/fink/path-prefix-clang/cc  -DCHECK_FUNCTION_EXISTS=xdr_uint64_t -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/sw/lib  CMakeFiles/cmTC_3ad6e.dir/CheckFunctionExists.c.o -o cmTC_3ad6e  -ltirpc 
ld: library not found for -ltirpc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
tirpc is just under a 'unix' conditional and was added post-v1.0.0.

**Comment by dmacks** [comment]
On 10.13 with Xcode10.1,
```
-- Found Threads: TRUE
-- Looking for xdr_uint64_t in tirpc
-- Looking for xdr_uint64_t in tirpc - not found
Strickly enforce PLPLOTDIR=/sw/lib/plplot
```
I have /usr/include/rpc/rpc.h that is part of a sublibrary of libSystem; did its header get pushed into an SDK subdir for you? Does the following compile if you give it...some -I to find it?
```
#include <stdlib.h>
#include <rpc/rpc.h>

int main(void) {
  xdr_int64_t((XDR*)NULL, NULL);
  exit(0);
}
```

**Comment by nieder** [comment]
```
nieder $ locate rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/WatchSimulator.platform/Developer/SDKs/WatchSimulator.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/usr/include/rpc/rpc.h
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/include/rpc/rpc.h
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/rpc/rpc.h
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/rpc/rpc.h
nieder $ 
```
But the above code does compile cleanly `cc foo.c -o foo` but does not link to -ltirpc (and segfaults when run).
```
nieder $ locate tirpc
nieder $
```

**Comment by dmacks** [comment]
That was just a hacky symbol-check via the compiler, not to run. It seems like cmake correctly diagnoses we don't need -ltirpc to get that symbol (passes test). But cmake finds the header for me and not for you, so it fails overall for you. Guess gdl needs to pass a -I or INCLUDE flag. Or is it a general cmake weakness not looking in the local platform's SDK?

**Comment by nieder** [comment]
Isn't my cmake failure in finding the library?
```
ld: library not found for -ltirpc
```

**Comment by dmacks** [comment]
> Isn't my cmake failure in finding the library?
> 
> ```
> ld: library not found for -ltirpc
> ```

Yes. But it doesn't require any -l flag on OS X. Cmake appears to check without the flag also, but unlike autoconf doesn't report that sort of success. Given my code-snippet compiled for you, which uses the symbol the cmake test checks the flag isn't needed (try `nm -m a.out | grep xdr_int64_t).

**Comment by nieder** [comment]
Adding `-DRPCDIR=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr \` lets configure finish, but still reports `not found`:

```
-- Found Threads: TRUE  
-- Looking for xdr_uint64_t in tirpc
-- Looking for xdr_uint64_t in tirpc - not found
Strickly enforce PLPLOTDIR=/sw/lib/plplot
```
But then there's lots of failures in `src/CFMTLexer.cpp`:
```
cd /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/build/src && /sw/var/lib/fink/path-prefix-clang/c++ -DHAVE_CONFIG_H -I/sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src -I/sw/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/sw/opt/hdf5.v1.10/include -I/sw/include/eigen3 -I/sw/build.build/gdl-1.0.1-1/gdl-1.0.1 -I/sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/antlr -I/sw/build.build/gdl-1.0.1-1/gdl-1.0.1/build -MD -O3 -DNDEBUG -fPIE -std=gnu++11 -MD -MT src/CMakeFiles/gdl.dir/CFMTLexer.cpp.o -MF CMakeFiles/gdl.dir/CFMTLexer.cpp.o.d -o CMakeFiles/gdl.dir/CFMTLexer.cpp.o -c /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/CFMTLexer.cpp
In file included from /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/CFMTLexer.cpp:3:
In file included from /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/includefirst.hpp:49:
In file included from /sw/include/eigen3/Eigen/Core:15:
/sw/include/eigen3/Eigen/src/Core/util/DisableStupidWarnings.h:48:38: warning: unknown warning group '-Wimplicit-int-float-conversion', ignored [-Wunknown-warning-option]
    #pragma clang diagnostic ignored "-Wimplicit-int-float-conversion"
                                     ^
In file included from /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/CFMTLexer.cpp:3:
In file included from /sw/build.build/gdl-1.0.1-1/gdl-1.0.1/src/includefirst.hpp:49:
In file included from /sw/include/eigen3/Eigen/Core:19:
In file included from /sw/include/eigen3/Eigen/src/Core/util/Macros.h:679:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:318:9: error: no member named 'signbit' in the global namespace
using ::signbit;
      ~~^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:319:9: error: no member named 'fpclassify' in the global namespace
using ::fpclassify;
      ~~^
```

The following fixes both issues and I get a successful build:
```diff
 		$FINK_CMAKE_ARGS \
-		-DCMAKE_CXX_FLAGS:STRING="-MD" \
+		-DCMAKE_CXX_FLAGS:STRING="-MD -isysroot ${sdk_path}" \
+		-DRPCDIR=${sdk_path}/usr \
 		-DEIGEN3=ON \
 		-DFFTW=ON \
```
c.f. https://stackoverflow.com/questions/58628377/catalina-c-using-cmath-headers-yield-error-no-member-named-signbit-in-th

**Comment by nieder** [comment]
fixed by 07f5f4e15f087189b12c4cba5c069305be070a68
